arbos/l2pricing/model.go

```diff
@@ -347,3 +350,47 @@ func (ps *L2PricingState) calcBaseFeeFromExponent(exponent arbmath.Bips) (*big.I
 		return minBaseFee, nil
 	}
 }
+
+func (ps *L2PricingState) MultiDimensionalPriceForRefund(gasUsed multigas.MultiGas) (*big.Int, error) {
+	if ps.ArbosVersion < ArbosMultiGasConstraintsVersion {
+		baseFeeWei, err := ps.BaseFeeWei()
+		if err != nil {
+			return nil, err
+		}
+
+		return new(big.Int).Mul(
+			new(big.Int).SetUint64(gasUsed.SingleGas()),
+			baseFeeWei,
+		), nil
+	}
+
+	total := new(big.Int)
+	minBaseFeeWei, err := ps.MinBaseFeeWei()
+	if err != nil {
+		return nil, err
+	}
+
+	for kind := range multigas.ResourceKind(multigas.NumResourceKind) {
+		baseFee, err := ps.multiGasBaseFees.Get(kind)
+		if err != nil {
+			return nil, err
+		}
+		// Override zero with minimum base fee,
+		// it may happen if there is no constraint configured for this resource kind
```

Comment:
This could not happen unless it is a bug. Also, it is better to set the default as the baseFee instead of the minBaseFee, so we don't give free refunds in case there is a bug.